java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java

```diff
@@ -28,29 +41,36 @@ public String version() {
 
   @Override
   public void start(Map<String, String> props) {
-    log.info("Task initialising");
-    printAsciiHeader(jarManifest, "/azure-eventhubs-ascii.txt");
-
-    Map<String, String> configProps;
-    boolean contextConfigEmpty = ofNullable(context).flatMap(
-        context -> ofNullable(context.configs()))
-      .flatMap(configs -> Optional.of(configs.isEmpty())).orElse(false);
-    if (contextConfigEmpty) {
-      configProps = props;
-    } else {
-      assert context != null;
-      configProps = context.configs();
-    }
+    eventHubsKafkaConsumerController = new EventHubsKafkaConsumerController(
+        new AzureEventHubsConfig(props), new BlockingQueueProducerProvider(),
+        new ArrayBlockingQueue<>(RECORDS_QUEUE_DEFAULT_SIZE));
+    ofNullable(this.context).flatMap(context -> ofNullable(context.offsetStorageReader()))
+        .ifPresent(TopicPartitionOffsetProvider::initialize);
+    initialize(eventHubsKafkaConsumerController);
+  }
+
+  /**
+   * Initializes the Task. This method shouldn't be called if start() was already called with
+   * {@link EventHubsKafkaConsumerController} instance.
+   *
+   * @param eventHubsKafkaConsumerController {@link EventHubsKafkaConsumerController} for this task
+   */
+  public void initialize(EventHubsKafkaConsumerController eventHubsKafkaConsumerController) {
```

Comment:
Is this meant to help with testing?

Reply:
Yes, this is to ommit `context()` calls that are part of `start()` method in tests.